zilliqa/src/consensus.rs

```diff
@@ -572,16 +596,26 @@ impl Consensus {
                 );
             }
 
-            if !block.committee.iter().any(|v| v.peer_id == self.peer_id()) {
-                trace!(
+            // Get possibly updated list of stakers
+            let Ok(stakers) = self.state.get_stakers_at_block(&block) else {
+                return Ok(None);
```

Comment:
Agreed that we shouldn't swallow this error.

Reply:
From reading `get_stakers_at_block` I think the only time it would return an `Err` is if something was fatally wrong (e.g. we weren't able to read data from our disk because someone drove their forklift through it). In these cases we should definitely propagate the error upwards. I think the same applies for the other cases Theo pointed out too.